experimental/ixdtf/src/parser.rs

```diff
@@ -2,6 +2,8 @@
 // called LICENSE at the top level of the ICU4X source tree
 // (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
 
+use std::collections::HashMap;
```

Comment:
Observation: this crate is `#![no_std]`, so you can't include things from `std`. This should be a build failure. However, I noticed that this crate isn't building at all in CI?

https://github.com/unicode-org/icu4x/pull/2786

Reply:
Now that https://github.com/unicode-org/icu4x/pull/2786 is merged, please merge main into this PR or rebase this PR on top of main.

Reply:
Supporting tags would need more designs because I don't have a good idea to move the pointer forward once to determine time zone vs tag in the parsed string.

So I created https://github.com/unicode-org/icu4x/pull/2821 to support time zone only first.